statchatui/src/Compare/Compare.jsx

```diff
@@ -105,6 +142,31 @@ const Compare = ({ signedInUserData }) => {
           </div>
         </div>
       )}
+      <div className="user-queue">
+        {userQueue.map((user, index) => (
+          <div key={index} className="user-item">
+            <h1
+              className="previous-user"
+              onMouseOver={() => setShowStats(index)}
+              onMouseOut={() => setShowStats(null)}
+              onClick={() => {
+                compareUsers(user, signedInUserData);
+              }}
+            >
+              {user.gamertag}
+            </h1>
+            <div className={`stats ${showStats == index ? "" : "hidden"}`}>
```

Comment:
If you don't want to mess with indices, you can refactor this userQueue piece into its own react component. (it will also make the code a bit nicer!)